effekt/shared/src/main/scala/effekt/Typer.scala

```diff
@@ -1607,6 +1607,14 @@ trait TyperOps extends ContextOps { self: Context =>
       }
     }
 
+  private[typer] def annotateSymbol(id: source.IdRef, sym: symbols.Symbol) =
+    annotations.update(Annotations.Symbol, id, sym)
+
+  override def symbolOption(id: source.Id): Option[Symbol] = id match {
+    case x: source.IdRef => Context.annotations.get(Annotations.Symbol, x).orElse { super.symbolOption(id) }
```

Comment:
What is the reason for the `IdRef` check?

Reply:
Mostly that I know that we just annotate those locally. We could drop it now (I originally had a separate annotation).

Reply:
Honestly, I don't know what's better, but I think not performing the check is conceptually simpler

Reply:
Removing it.